models/aeif_cond_alpha.cpp

```diff
@@ -167,6 +167,16 @@ nest::aeif_cond_alpha::State_::State_( const State_& s )
   }
 }
 
+nest::aeif_cond_alpha::State_& nest::aeif_cond_alpha::State_::operator=( const State_& s )
+{
+  r_ = s.r_;
+  for ( size_t i = 0; i < STATE_VEC_SIZE; ++i )
+  {
+    y_[ i ] = s.y_[ i ];
+  }
+  return *this;
+}
+
```

Comment:
This re-adds `operator=()` that was removed in #2042 (for example [here](https://github.com/nest/nest-simulator/pull/2042/files#diff-7572728594d53870157ce7b8b13f1122ec6a1bde22908071f022dfb70c2a1e04L170)). Why?

The same comment applies to many of the changes below.

@stinebuu: what do you think about this?

Reply:
@jougs @hakonsbm @stinebuu I think we actually need the assignment operators for `Parameters_` and `State_` for all models to support this pattern:
```c++
inline void
aeif_psc_alpha::set_status( const DictionaryDatum& d )
{
  Parameters_ ptmp = P_;     // temporary copy in case of errors
  ptmp.set( d, this );       // throws if BadProperty
  State_ stmp = S_;          // temporary copy in case of errors
  stmp.set( d, ptmp, this ); // throws if BadProperty
  ArchivingNode::set_status( d );
  P_ = ptmp;
  S_ = stmp;
}
```

I fear we overlooked this with #2042, good to have caught this (my bad ...). This also seems to indicate that we have no tests picking up on this. For models with reasonably plain or smart members, I suppose the automatically constructed assignment operators do the trick, but not for those with C-style arrays as this one. But we should we rely on default operators even where they work?